torch_geometric/datasets/ba_shapes.py

```diff
@@ -0,0 +1,103 @@
+from typing import Optional, Callable
+
+import torch
+import numpy as np
+import networkx as nx
+from torch_geometric.data import InMemoryDataset, Data
+
+
+def house():
+    G = nx.house_graph()
+    labels = [1, 1, 2, 2, 3]
+    attr = {i: labels[i] for i in range(5)}
+    nx.set_node_attributes(G, attr, 'label')
+    nx.set_edge_attributes(G, 1, 'label')
+    return G
+
+
+def ba(num_nodes=300, m=5, seed=None):
+    G = nx.barabasi_albert_graph(num_nodes, m, seed=seed)
+    nx.set_node_attributes(G, 0, 'label')
+    nx.set_edge_attributes(G, 0, 'label')
+    return G
+
+
+class BAShapes(InMemoryDataset):
+    r"""The BA-Shapes dataset from the `"GNNExplainer: Generating Explanations
+    for Graph Neural Networks" <https://arxiv.org/pdf/1903.03894.pdf>`_ paper,
+    containing a Barabasi-Albert (BA) graph with 300 nodes and a set of 80
+    "house"-structured graphs connected to it.
+
+    Args:
+        seed (int, optional): Random seed used for :obj:`networkx` Barabási
+            Albert graph generation and for :obj:`numpy` methods.
+            (default: :obj:`None`)
+        connection_distribution (string, optional): Specifies how the houses
+            and the BA graph get connected. Valid inputs are :obj:`random`
+            (random BA graph nodes are selected for connection to the houses),
+            :obj:`uniform` (uniformly distributed BA graph nodes are selected
+            for connection to the houses). (default: :obj:`random`)
+        transform (callable, optional): A function/transform that takes in an
+            :obj:`torch_geometric.data.Data` object and returns a transformed
+            version. The data object will be transformed before every access.
+            (default: :obj:`None`)
+    """
+    def __init__(
+        self,
+        seed: Optional[int] = None,
+        connection_distribution: str = 'random',
+        transform: Optional[Callable] = None,
+    ):
+        super().__init__('.', transform, None, None)
+        assert connection_distribution in ['random', 'uniform']
+        if seed is not None:
+            np.random.seed(seed)
```

Comment:
I think I would drop this line. If the user wants deterministic behavior, the user can simply set `torch_geometric.seed_everything` at the root-level.